app/assets/javascripts/camaleon_cms/admin/_libraries.js

```diff
@@ -1,136 +1,168 @@
+          $contentImage.show()
+        } else
+          $contentImage.hide()
+      }).trigger('change')
+    })
+  }
 
-    // create inline input file uploader with an icon to upload file
-    // options: all options needed for uploader
-    // you can add an attribute "data-format" in the input to define the file formats required
-    $.fn.input_upload_field = function (options) {
-        this.each(function () {
-            var input = $(this);
-            var def = {
-                formats: (input.attr("data-format") || "image"),
-                selected: function (file) {
-                    input.val(file.url);
-                }
-            };
-            if (!input.parent().hasClass("input-group")) {
-                input.wrap('<div class="group-input-fields-content input-group"></div>');
-                input.after('<span class="input-group-addon btn_upload"><i class="fa fa-upload"></i> </span>');
-                input.addClass("form-control");
-            }
-            input.next("span").click(function () {
-                $.fn.upload_filemanager($.extend({}, def, (options || {})));
-            });
-        });
-    }
-})(jQuery);
+  // create inline input file uploader with an icon to upload file
+  // options: all options needed for uploader
+  // you can add an attribute "data-format" in the input to define the file formats required
+  $.fn.input_upload_field = function(options) {
+    this.each(function() {
+      const input = $(this)
+      const def = {
+        formats: (input.attr('data-format') || 'image'),
+        selected: function(file) {
+          input.val(file.url)
+        }
+      }
+      if (!input.parent().hasClass('input-group')) {
+        input.wrap('<div class="group-input-fields-content input-group"></div>')
+        input.after('<span class="input-group-addon btn_upload"><i class="fa fa-upload"></i> </span>')
+        input.addClass('form-control')
+      }
+      input.next('span').click(function() {
+        $.fn.upload_filemanager($.extend({}, def, (options || {})))
+      })
+    })
+  }
+})(jQuery)
 
 // serialize form into json
-!(function ($) { jQuery.fn.serializeObject = function () { var json = {}; jQuery.map(jQuery(this).serializeArray(), function (n, i) { var __i = n.name.indexOf('['); if (__i > -1) { var o = json; _name = n.name.replace(/\]/gi, '').split('['); for (var i = 0, len = _name.length; i < len; i++) { if (i == len - 1) { if (o[_name[i]] && $.trim(_name[i]) == '') { if (typeof o[_name[i]] == 'string') { o[_name[i]] = [o[_name[i]]]; } o[_name[i]].push(n.value); } else o[_name[i]] = n.value || ''; } else o = o[_name[i]] = o[_name[i]] || {}; } } else { if (json[n.name] !== undefined) { if (!json[n.name].push) { json[n.name] = [json[n.name]]; } json[n.name].push(n.value || ''); } else json[n.name] = n.value || ''; } }); return json; }; })(jQuery);
+// eslint-disable-next-line no-unused-expressions
+!(function($) {
```

Comment:
I am in doubt of this serialization function. It doesn't seem to work with nested forms (which is correct, as nested forms are not considered valid HTML). As for the flat forms, the serialization is currently said to be possible with lesser efforts using the `Object.fromEntries()` method  - see https://stackoverflow.com/a/61802665/4812102

Reply:
I'm not familiar with these issues. It probably only matters if we're going to keep jQuery, since otherwise it will get rewritten anyway.

Reply:
Well, it would be great to be simplified at least until we're using jQuery.